docs/manual/non-empty-checker.tex

```diff
@@ -18,6 +18,12 @@
   javac -processor org.checkerframework.checker.nonempty.NonEmptyChecker \emph{MyJavaFile}.java
 \end{alltt}
 
+The Non-Empty Checker has a second modality, in which it mostly acts like
```

Comment:
We don't want this to be very prominent.  I suggest moving this to the very end of the chapter, in its own subsection.  You can remark that this is what the OptionalChecker does.

Are there any issues with running both the (new) OptionalChecker and the Non-Empty Checker at the same time?  If so, please discuss.